Error building wiht VSCode devcontainer

I got this error building with the VSCode container

```
Compiling [arm64-shared] test/Slice/escape/Client.cpp
test/Slice/escape/Client.cpp: In function 'void testtypes(const Ice::CommunicatorPtr&)':
test/Slice/escape/Client.cpp:124:17: error: array subscript -6 is outside array bounds of 'int (* [1152921504606846975])(...)' [-Werror=array-bounds]
  124 |     g->_cpp_case(0, d2);
```

Commenting out the following lines solve the issue

https://github.com/zeroc-ice/ice/blob/bf991f4f7604869517f9211f18a6048239521551/cpp/test/Slice/escape/Client.cpp#L124-L125

--- a/cpp/test/Slice/escape/Client.cpp
+++ b/cpp/test/Slice/escape/Client.cpp
@@ -121,8 +121,14 @@ void testtypes(const Ice::CommunicatorPtr& communicator)
     _cpp_and::switchPtr f1 = std::make_shared<switchI>();
 
     _cpp_and::doPrxPtr g;
+
+// Work-around for:
+// error: array subscript -6 is outside array bounds of ‘int (* [1152921504606846975])(...)’ [-Werror=array-bounds]
+// apparently caused by https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105523 and fixed in GCC 13.3.
+#if !defined(NDEBUG) || !defined(__GNUC__) || (__GNUC__ > 13) || (__GNUC__ == 13 && __GNUC_MINOR__ >= 3)
     g->_cpp_case(0, d2);
     g->_cpp_explicit();
+#endif
     _cpp_and::doPtr g1 = std::make_shared<doI>();
 
     _cpp_and::_cpp_extern h;

--- a/cpp/test/Slice/escape/Client.cpp
+++ b/cpp/test/Slice/escape/Client.cpp
@@ -124,10 +124,14 @@ void testtypes(const Ice::CommunicatorPtr& communicator)
 
 // Work-around for:
 // error: array subscript -6 is outside array bounds of ‘int (* [1152921504606846975])(...)’ [-Werror=array-bounds]
-// apparently caused by https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105523 and fixed in GCC 13.3.
-#if !defined(NDEBUG) || !defined(__GNUC__) || (__GNUC__ > 13) || (__GNUC__ == 13 && __GNUC_MINOR__ >= 3)
+#if defined(NDEBUG) && defined(__GNUC__)
+#   pragma GCC diagnostic push
+#   pragma GCC diagnostic ignored "-Warray-bounds"
+#endif
     g->_cpp_case(0, d2);
     g->_cpp_explicit();
+#if defined(NDEBUG) && defined(__GNUC__)
+#   pragma GCC diagnostic pop
 #endif
     _cpp_and::doPtr g1 = std::make_shared<doI>();
 

